measure_innersource.py

```diff
@@ -18,6 +18,108 @@
 from markdown_writer import write_to_markdown
 
 
+class CaseInsensitiveDict:
+    """A dictionary wrapper that performs case-insensitive key lookups.
+
+    This class wraps a dictionary and provides case-insensitive access to its values
+    while preserving the original data structure. It's used to handle GitHub usernames
+    which should be matched case-insensitively.
+
+    Note: If the input dictionary contains keys that differ only in case (e.g., 'User'
+    and 'user'), a ValueError will be raised during initialization to prevent ambiguity
+    and data loss.
+    """
+
+    def __init__(self, data):
+        """Initialize the case-insensitive dictionary.
+
+        Args:
+            data: A dictionary to wrap with case-insensitive key access
+
+        Raises:
+            ValueError: If the dictionary contains duplicate keys that differ only in case
+        """
+        self._data = data
+        # Create a lowercase key mapping to original keys
+        self._key_map = {}
+        for key in data.keys():
```

Comment:
Not sure how I feel about this custom class. A simpler version could be 


```python
class CaseInsensitiveDict(dict):
    def __setitem__(self, key, value):
        super().__setitem__(key.lower(), value)

    def __getitem__(self, key):
        return super().__getitem__(key.lower())

    def __contains__(self, key):
        return super().__contains__(key.lower())
```

It does assume the keys are already lowercase.

or we could use `requests.structures.CaseInsensitiveDict` (would require adding `requests` as a dependency though).

```python
from requests.structures import CaseInsensitiveDict

ci_dict = CaseInsensitiveDict()
ci_dict["Key0"] = "a"
ci_dict["KEY1"] = "b"

print(ci_dict["key1"])  # Output: Value1
```

wdyt? Do we really want to maintain this? I'm good if you are.

Reply:
I think adding the requests dependency is ok especially over maintaining a custom class. Thanks for the suggestion. Hadn't thought of an alternative here.

Reply:
implemented at f124b1b